clouway-pcache-client-redis/src/main/java/com/clouway/api/pcache/extensions/redis/RedisCacheManager.java

```diff
@@ -0,0 +1,253 @@
+package com.clouway.api.pcache.extensions.redis;
+
+import com.clouway.api.pcache.CacheException;
+import com.clouway.api.pcache.CacheManager;
+import com.clouway.api.pcache.MatchResult;
+import com.clouway.api.pcache.NamespaceProvider;
+import com.clouway.api.pcache.SafeValue;
+import com.clouway.api.pcache.extensions.redis.RedisFormat.ValueAndFlags;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Transaction;
+import redis.clients.jedis.params.SetParams;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+class RedisCacheManager implements CacheManager {
+  private static final int DEFAULT_TIMEOUT_SECONDS = 3000;
+  private final Jedis jedis;
+  private final NamespaceProvider namespaceProvider;
+
+  RedisCacheManager(Jedis jedis, NamespaceProvider namespaceProvider) {
+    this.jedis = jedis;
+    this.namespaceProvider = namespaceProvider;
```

Comment:
I prefer to have the namespace as param in the methods

Reply:
This will change the whole interface and will enforce code changes in all client apps.